h5py/_hl/selections.py

```diff
@@ -91,6 +91,68 @@ def select(shape, args, dsid):
     return sel
 
 
+class MultiBlockSlice(object):
+
+    """
+        A conceptual extension of the built-in slice object to allow selections
+        using start, stride, count and block.
+
+        If given, these parameters will be passed directly to
+        H5Sselect_hyperslab. The defaults are start=0, stride=1, block=1,
+        count=length, which will select the full extent.
+
+        __init__(start, count, stride, block) => Create a new MultiBlockSlice, storing
+            any given selection parameters and using defaults for the others
+        start => The offset of the starting element of the specified hyperslab
+        count => The number of blocks to select
+        stride => The number of elements between the start of one block and the next
+        block => The number of elements in each block
+
+    """
+
+    def __init__(self, start=0, count=None, stride=1, block=1):
+        if start < 0:
+            raise ValueError("Start must be positive")
+        if count is not None and count < 1 or stride < 1 or block < 1:
+            raise ValueError("Count, block and stride must be non-zero")
+        if block > stride:
+            raise ValueError("Blocks will overlap if block > stride")
+
+        self.start = start
+        self.count = count
+        self.stride = stride
+        self.block = block
+
+    def indices(self, length):
+        """Calculate and validate start, stop, stride and block for the given length"""
+        if self.count is None:
+            # Select as many full blocks as possible without exceeding extent
+            count = (length - self.start - self.block) // self.stride + 1
+            if count < 1:
+                raise ValueError(
+                    "No full blocks can be selected using {} "
+                    "on dimension of length {}".format(self._repr(), length)
+                )
+        else:
+            count = self.count
+
+        end_index = self.start + self.block + (count - 1) * self.stride - 1
+        if end_index >= length:
+            raise ValueError(
```

Comment:
This should never happen if count is calculated, right? The other error would catch it in that case. No problem to have an extra sanity check, I'm just making sure that I'm understanding the code correctly.

Reply:
Yeah, this check is for when count is user defined, but it is invalid for the length of the dataset the selection is applied to (this case is hit in `test_block_overruns_extent_error`). The block could be indented into the else of the previous block, but I didn't think there was any reason to. Plus, I guess it is possible that in future the count calculation is updated and broken and it would still be caught in that case.

Reply:
Thanks for confirming. Let's leave it as is.